### PR TITLE
Add source and destination workload_kind context labels (Hubble).

### PR DESCRIPTION
**Pull request by marqc**
Please ensure your pull request adheres to the following guidelines:

- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Add source and destination workload_kind context labels (Hubble).

```release-note
Add source and destination workload_kind context labels (Hubble).
```

### CODE REVIEW

File: Documentation/observability/metrics.rst
Comment (zacharysarah) [inline Documentation/observability/metrics.rst]: No Latin abbreviations
```suggestion
``source_workload_kind``       The kind of the source pod's workload, for example, Deployment, Statefulset, Daemonset, ReplicationController, CronJob, Job, DeploymentConfig (OpenShift).
```

File: Documentation/observability/metrics.rst
Comment (zacharysarah) [inline Documentation/observability/metrics.rst]: No Latin abbreviations
```suggestion
``destination_workload_kind``  The kind of the destination pod's workload, for example, Deployment, Statefulset, Daemonset, ReplicationController, CronJob, Job, DeploymentConfig (OpenShift).
```

**Review by zacharysarah — APPROVED** [review]
@marqc Good work. ✨ Some minor changes, otherwise LGTM. Approving with the understanding that changes are required prior to merge.

**Review by nebril — COMMENTED** [review]
Thanks for the PR! The code itself looks good to me.

I am a bit concerned with whether this change would increase cardinality of the metric significantly (which in itself already has a lot of labels). I would like to defer to @chancez to make a judgement here.

**Comment by marqc** [comment]
> Thanks for the PR! The code itself looks good to me.
> 
> I am a bit concerned with whether this change would increase cardinality of the metric significantly (which in itself already has a lot of labels). I would like to defer to @chancez to make a judgement here.

It should not unless you have distinct workload kinds with the same workload name (ie. cronjob, statefulset and deployment named "mysql"). But if you are in this situation, you probably want to have this additional label to be able to distinguish metrics of these workloads.

Additionally you need to enable these context labels in metrics config, so it would not change the cardinality without user action. Unless you want me to change default metrics settings in helm charts as well?

**Review by chancez — APPROVED** [review]
LGTM. Cardinality shouldn't be much different unless you name all your workloads the same across different kinds, which is uncommon, and this is opt-in as well.

**Review by kaworu — APPROVED** [review]

**Comment by kaworu** [comment]
@marqc [checkpatch is unhappy](https://github.com/cilium/cilium/actions/runs/5209795928/jobs/9400084652?pr=25715#step:3:36), could you please update the commit signed-off-by to

```
Signed-off-by: Marek Chodor <mchodor@google.com>
```

**Comment by ldelossa** [comment]
@marqc - this PR needs to wait until release to merge. But I am noticing there are two stuck tests. 

Can you rebase this PR on `main` preemptively? That way we can run tests and get it all set to merge once the upcoming release is done.

**Comment by ldelossa** [comment]
Travis CI seems to be stuck here. I'm going to attempt to close and open this PR. GH had some downtime yesterday, and I suspect this caused an issue with our CI events.

**Comment by ldelossa** [comment]
@marqc 

Sorry to ask you to do this, but it appears all your PRs are having trouble triggering TravisCI. 
I tried to see if I can open a PR myself with your fork, but GH does not recognize your fork when we attempt to 'compare across forks'. 

I'm going to leave this message on your other PRs as well, but can you please close this PR and open a new one with the same changes? Hopefully this will fix these issues.

**Comment by marqc** [comment]
@ldelossa I've been on vacation for the last few weeks. Right now I rebased my commit on top of the main branch. If the problem persists I will close this PR and reopen a new one.

**Pull request closed without merging**
